### PR TITLE
[Bug-fix] Decimal Divide, Mod Zero Result Should be NULL.

### LINKED ISSUE #6049: [Bug] Decimal Value Divide, Mod Zero Result Should Be Null
**Describe the bug**
Now, Decimal Divide or Mod zero doris will get 0, but not null.

```
select 10.2 / 0.0;
+------------+
| 10.2 / 0.0 |
+------------+
|          0 |
+------------+
```

But in Mysql, no metter `int` or `decimal`, divide, mod zero always get Null
```
select 10.2 % 0.0;
+------------+
| 10.2 % 0.0 |
+------------+
|       NULL |
+------------+
```

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

RT

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6049) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/runtime/decimalv2_value.cpp [hunk 1]
```diff
@@ -192,7 +192,7 @@ DecimalV2Value operator/(const DecimalV2Value& v1, const DecimalV2Value& v2) {
     int128_t x = v1.value();
     int128_t y = v2.value();
 
-    //todo: return 0 for divide zero
+    DCHECK(y != 0);
     if (x == 0 || y == 0) return DecimalV2Value(0);
```
Comment (morningman) [inline be/src/runtime/decimalv2_value.cpp:196]:
```suggestion
    if (x == 0) return DecimalV2Value(0);
```

File: be/src/runtime/decimalv2_value.cpp [hunk 2]
```diff
@@ -207,7 +207,7 @@ DecimalV2Value operator%(const DecimalV2Value& v1, const DecimalV2Value& v2) {
...
```
Comment (morningman) [inline be/src/runtime/decimalv2_value.cpp:211]:
```suggestion
    if (x == 0) return DecimalV2Value(0);
```

File: be/src/runtime/decimalv2_value.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/runtime/decimalv2_value.cpp:196]: we have to do this check to prevent divide / zero which will cause core in BE.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**